repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -4721,6 +4713,54 @@ private void updateDomainAttribute(AtlasVertex vertex, Set<String> existingValue
         existingValues.forEach(existingValue -> vertex.removePropertyValue(DOMAIN_GUIDS_ATTR, existingValue));
         vertex.setProperty(DOMAIN_GUIDS_ATTR, meshEntityId);
     }
+    public AtlasVertex moveBusinessPolicy(Set<String> policyIds, String assetId, String type) throws AtlasBaseException {
+        // Retrieve the AtlasVertex for the given assetId
+        AtlasVertex assetVertex = AtlasGraphUtilsV2.findByGuid(graph, assetId);
+
+        if(assetVertex == null){
+            throw new AtlasBaseException(AtlasErrorCode.INVALID_PARAMETERS, "assetId not found");
+        }
+
+        // Get the sets of governed and non-compliant policy GUIDs
+        Set<String> governedPolicies = assetVertex.getMultiValuedSetProperty(ASSET_POLICY_GUIDS, String.class);
+        Set<String> nonCompliantPolicies = assetVertex.getMultiValuedSetProperty(NON_COMPLIANT_ASSET_POLICY_GUIDS, String.class);
+
+        // Determine if the type is governed or non-compliant and move policies accordingly
+        boolean isGoverned = MoveBusinessPolicyRequest.Type.GOVERNED.getDescription().equals(type);
+
+        policyIds.forEach(policyId -> {
+            if (isGoverned) {
+                assetVertex.setProperty(ASSET_POLICY_GUIDS, policyId);
```

Comment:
What if the policy was already in the ASSET_POLICY_GUIDS array? this will create a duplicate entry in ASSET_POLICY_GUIDS attribute right?